packages/otelbin/src/middleware.test.ts

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it, jest, afterEach } from "@jest/globals";
+import { NextRequest, NextResponse } from "next/server";
+import { handleShortLinkRequest } from "./middleware";
+
+const rewriteSpy = jest.spyOn(NextResponse, "rewrite");
+const redirectSpy = jest.spyOn(NextResponse, "redirect");
+
+jest.mock("@upstash/redis/nodejs", () => {
+	return {
+		Redis: {
+			fromEnv: jest.fn(() => {
+				return {
+					get: jest.fn(() => {
+						return "https://www.otelbin.io";
+					}),
+				};
+			}),
+		},
+	};
+});
+
+describe("handleShortLinkRequest", () => {
+	afterEach(() => {
+		redirectSpy.mockReset();
+		rewriteSpy.mockReset();
+	});
+
+	it("If the 'bot' URL parameter is set to true, it should response with rewrite the URL to '/social-preview/${shortLink}' to generate the open graph image.", async () => {
+		const mockRequest = new Request("https://otelbin.io/s/1d643f1e-fd9b-4123-948c-f374d2fc923e?bot=true");
+		const mockNextRequest = new NextRequest(mockRequest, {});
+		const result = await handleShortLinkRequest(mockNextRequest);
+		expect(redirectSpy).toHaveBeenCalledTimes(0);
+		expect(rewriteSpy).toHaveBeenCalledTimes(1);
```

Comment:
**Suggestion:** Test the arguments to `rewriteSpy` to ensure that the correct rewrite is executed.